core/src/test/scala/integration/kafka/server/DynamicBrokerReconfigurationTest.scala

```diff
@@ -429,6 +430,20 @@ class DynamicBrokerReconfigurationTest extends QuorumTestHarness with SaslSetup
       verifyProduceConsume(producer, consumer, 10, topic)
     }
 
+    def verifyBrokerToControllerCall(controller: KafkaServer): Unit = {
+      val nonControllerBroker = servers.find(_.config.brokerId != controller.config.brokerId).get
+      val brokerToControllerManager = nonControllerBroker.clientToControllerChannelManager
+      val completionHandler = new TestControllerRequestCompletionHandler()
+      brokerToControllerManager.sendRequest(new MetadataRequest.Builder(new MetadataRequestData()), completionHandler)
+      TestUtils.waitUntilTrue(() => {
+        completionHandler.completed.get() || completionHandler.timedOut.get()
+      }, "Timed out while waiting for broker to controller API call")
+      val response = completionHandler.actualResponse.getOrElse(throw new IllegalStateException("No response recorded even though request is completed"))
```

Comment:
throwing exception in the test seems not a good practice. Maybe we can change to this:
```
assertTrue(completionHandler.actualResponse.isDefined, "No response recorded even though request is completed")
val response = completionHandler.actualResponse.get
```

Reply:
Also, I think we should fail when timeout, so that we know what happened in the request. Ex:
```
assertTrue(completionHandler.timedOut.isEmpty, "broker to controller request is timeout")
```
WDYT?

Reply:
Both your comments make sense. I have make the changes as per your suggestions.